### PR TITLE
Bug 1694522 - Do not compare UUIDs when we dont provide them

### PR DESCRIPTION
**Pull request by ewolinetz**
Since CLO does not specify a UUID for the nodes for EO, we need to ignore that the GenUUID field has changed on the CR when doing a comparison to see if changes are required to be made.

Also required to update vendor to get GenUUID field in api.

Ref: https://bugzilla.redhat.com/show_bug.cgi?id=1694522

### CODE REVIEW

File: pkg/k8shandler/logstore.go [hunk 1]
```diff
@@ -120,7 +123,6 @@ func (cluster *ClusterLogging) newElasticsearchCR(elasticsearchName string) *ela
 		Roles:        []elasticsearch.ElasticsearchNodeRole{"client", "data", "master"},
 		NodeCount:    cluster.Spec.LogStore.NodeCount,
 		NodeSelector: cluster.Spec.LogStore.NodeSelector,
-		Resources:    *resources,
```
Comment (jcantrill) [inline pkg/k8shandler/logstore.go]: Did we lose this?  It looks like we update to include a default cpuRequest.  Is `esNode` the common defaults?

Comment (ewolinetz) [reply pkg/k8shandler/logstore.go]: we didn't, but I'd rather us specify it just at the common node level than at both the individual node and the common node.

Comment (josefkarasek) [reply pkg/k8shandler/logstore.go]: +1 Bigly

File: pkg/k8shandler/logstore.go [hunk 2]
```diff
@@ -229,9 +231,98 @@ func isElasticsearchCRDifferent(current *elasticsearch.Elasticsearch, desired *e
...
+
+	// if we didn't find a role match, then that means changes were made
+	if !foundRoleMatch {
+		different = true
+	}
+
+	return desired, different
+}
+
+func areNodeRolesSame(lhs, rhs elasticsearch.ElasticsearchNode) bool {
```
Comment (josefkarasek) [inline pkg/k8shandler/logstore.go]: For the record, this is style which I prefer (like better)

```
    func getRole(roles []ElasticsearchNodeRole, roleType ElasticsearchNodeRole) bool {
        for _, role := range roles {
            if role == roleType {
                return true
            }
        }
        return false
    }
    // ...
    // not equal matters here
    if getRole(lhs.Roles, elasticsearch.ElasticsearchRoleClient) != getRole(rhs.Roles, elasticsearch.ElasticsearchRoleClient) {
        return false
    }
    // ...
    return true
```

**Comment by wking** [comment]
/hold

I dunno why the restest bot is picking on this PR, but it is, despite the fact that this won't go green until at least the conflicts are resolved.  I'm dropping a `/hold` to call the bot off until that rebase has happened, feel free to `/hold cancel` afterwards.

**Comment by richm** [comment]
Usually the needs-rebase label is added automatically - not sure why it wasn't added

**Comment by ewolinetz** [comment]
still checking vendored changes /hold

**Pull request merged**